src/validation/custom.validation.ts

```diff
@@ -698,3 +700,26 @@ export const checkFieldIfRadioButtonSelectedAndFieldsEmpty = (isPrimaryField: bo
   }
 };
 
+export const checkDateIsBeforeOrOnNextMadeUpToDate = async (req, dayStr: string, monthStr: string, yearStr: string): Promise<boolean> => {
+  const appData: ApplicationData = getApplicationData(req.session);
+  if (!appData.entity_number) {
+    logger.errorRequest(req, "checkDateBeforeNextMadeUpToDate validation - Unable to find entity number in application data.");
+    throw new Error(ErrorMessages.UNABLE_TO_RETRIEVE_ENTITY_NUMBER);
```

Comment:
A shame we can't get the 'service error page' to show here, as there's nothing the user can do to address this.

Reply:
I did have a quick look, i'll try and dig a bit further to see if there is a way of doing this

Reply:
i've re-jigged it so that the service error page will now show if unable to get the made up to date or entity number is missing